scripts/build/update-gh-pages.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash -e
+set -o pipefail
+
+script=`basename $0`
+
+usage () {
+cat << EOF
+Usage: $script [-h] [-a] [BUILD_SUBDIR]
+
+Options:
+  -h         show this help and exit
+  -a         amend (with --reset-author) instead of creating a new commit
+EOF
+}
+
+#
+# Argument parsing
+#
+while getopts "hap:" opt; do
+    case $opt in
+        h)  usage
+            exit 0
+            ;;
+        a)  amend="--amend --reset-author"
+            ;;
+        *)  usage
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND - 1))"
```

Comment:
minor nitpick: we consistently avoid using getopt in all of our other shell scripts (mostly because it's so limited). In the other scripts we use something like this with small variations:
```
while [ "${1#-}" != "$1" -a -n "$1" ]; do
    case "$1" in 
        -a|--amend)
            amend="--amend --reset-author"
            ;;
        -h|--help)
            usage
            exit 0
            ;;
        *)
            usage
            exit 1
            ;;
    esac
    shift
done
```

Reply:
I don't think the limited nature of getopt really matters in script like this not intended to be really executed by users. But, I'm quite indifferent here and don't have a particular affection towards getopt, either. Consistency is always good. I'll change this

Reply:
Changed